### DISCUSSION TITLE
How to slice a tensor without copying it?

### DISCUSSION THREAD

**Discussion by InAnYan** [tracel-ai/burn#2449] [Q&A] [answered]
Why `.slice()` takes a `self` argument?

Is there any variant that would make a new tensor that is a slice of an existing tensor but without copying the original tensor (e.g.: a method that accepts `&self`)?

**Comment by InAnYan** [comment]
How would I properly split into train and test tensors?

**Comment by laggui** [comment, marked as answer]
If you still want to use the tensor after the slice, simply do `tensor.clone().slice(...)`. Cloning does not copy the data it just increases a reference count. And then you can proceed to use `tensor` in following operations.

**Reply by InAnYan** [reply]
Oh, it's not cloning literally. I see, thanks!
